src/requestinterceptor.js

```diff
@@ -9,4 +9,22 @@ chrome.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
     console.log(tabId)
     console.log(changeInfo)
     console.log(tab)
+    const URL = tab.url.split('/')[2]
+    await chrome.storage.local.get(["blockedSites", "warnedSites"], async (storage) => {
+        if (storage.warnedSites.find(x => URL.includes(x)) || storage.blockedSites.find(x => URL.includes(x))) {
+            // close target
+            chrome.tabs.remove(tabId)
```

Comment:
Why not just update the tab to have a new URL?

https://developer.chrome.com/docs/extensions/reference/tabs/#method-update

Reply:
Hi TheENGR,

What is the advantage of not uploading a new tab?  If the "Man in the Middle" is created within the existing tab then we could run into issues with the browser history.

Reply:
We MIGHT be able to re-write the history

Wanted to not load jerks so they don't get paid for ADs

Reply:
https://developer.chrome.com/docs/extensions/reference/history/ 

We can re-wrote the history if needed